### PR TITLE
Refactor 'PRODUCTION' boolean environment variable to a key-value pair

### PR DESCRIPTION
**Pull request by gdiazc**
'APP_ENV' should be set to 'PRODUCTION' in production.

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -38,8 +39,14 @@ def get_tweets():
     client = Elasticsearch(hosts=[params['es_endpoint']])
     tweets = es_client.get_tweets(client)
 
-    xs = []
-    ys = []
+
+    import dateutil
```
Comment (dev1-fund3) [inline app.py]: move imports to top of file.

File: app.py [hunk 2]
```diff
@@ -38,8 +39,14 @@ def get_tweets():
...
+
+    id_strs = []
+    created_ats = []
+    full_texts = []
+    subjs = []
```
Comment (dev1-fund3) [inline app.py]: use more descriptive variable names. Is this subjects? and below polarities?

File: app.py [hunk 3]
```diff
@@ -51,17 +58,38 @@ def get_tweets():
...
+
+    df = pd.DataFrame({
+        'id_str': id_strs,
+        'created_at': created_ats,
+        'full_text': full_texts,
+        'subjectivity': subjs,
+        'polarity': polas
+    })
 
     fig = bokeh.plotting.figure(plot_width=600, plot_height=400)
```
Comment (dev1-fund3) [inline app.py]: create a separate method for plotting the dataframe

**Review by dev1-fund3 — CHANGES_REQUESTED** [review]

File: app.py (hunk 1 above)
Comment (gdiazc) [reply app.py]: Resolved in next commit!

File: app.py (hunk 2 above)
Comment (gdiazc) [reply app.py]: Resolved in next commit!

File: app.py (hunk 3 above)
Comment (gdiazc) [reply app.py]: Resolved in next commit!

File: bin/cron_pull_tweets.py [hunk 4]
```diff
@@ -67,4 +67,8 @@ def run_task(params, ntweets=3, maxpages=2):
         with open('secrets.txt', 'r') as f:
             params = json.load(f)
 
-    run_task(params)
+    ntweets = 100
```
Comment (dev1-fund3) [inline bin/cron_pull_tweets.py]: Assuming this is used for debugging? If so, make an env var

**Review by dev1-fund3 — COMMENTED** [review]
former concerns addressed. 1 question for clarification

File: bin/cron_pull_tweets.py (hunk 4 above)
Comment (gdiazc) [reply bin/cron_pull_tweets.py]: The env var 'P_CRON_NTWEETS' was created (see lines 71–74). Line 70 serves as a fallback default, although perhaps we could refactor that!

File: app.py [hunk 5]
```diff
@@ -1,24 +1,24 @@
...
-    return render_template('index.html', resources=bokeh.resources.CDN.render())
+nltk.download('punkt')
+app = Flask(__name__)
 
 
-@app.route('/get_tweets.json', methods=['GET'])
-def get_tweets():
-    if 'PRODUCTION' in os.environ:
+def _get_env_params():
+    if 'APP_ENV' in os.environ and os.environ['APP_ENV'] == 'PRODUCTION':
```
Comment (calvinleungyk) [inline app.py]: minor nitpick: I think you can provide a default value with `os.getenv('APP_ENV', 'default_value_of_something')` and so you don't need that `if 'APP_ENV' in os.environ` check

File: bin/cron_pull_tweets.py [hunk 6]
```diff
@@ -67,4 +67,8 @@ def run_task(params, ntweets=3, maxpages=2):
...
+    if 'P_CRON_NTWEETS' in os.environ:
```
Comment (calvinleungyk) [inline bin/cron_pull_tweets.py]: Same comment for `os.getenv()`

File: app.py [hunk 7]
```diff
@@ -1,24 +1,24 @@
 from flask import Flask, render_template
 import os
 import simplejson as json
```
Comment (calvinleungyk) [inline app.py]: Sort and format the imports as follows: http://google.github.io/styleguide/pyguide.html#313-imports-formatting

**Review by calvinleungyk — CHANGES_REQUESTED** [review]
Seems like you should add a yaml config file and read from yaml instead of checking env?  Looks like you have a couple instances of doing that and I anticipate more flags or envvars that you have to set

File: app.py (hunk 5 above)
Comment (gdiazc) [reply app.py]: Ahh yes, I should have known! I will most definitely use that in the next commit.

Comment (gdiazc) [reply app.py]: Ready!

File: bin/cron_pull_tweets.py (hunk 6 above)
Comment (gdiazc) [reply bin/cron_pull_tweets.py]: Incorporated! See [
f68b21a](https://github.com/fund3/sentiment-dashboard/pull/3/commits/f68b21a7325889dba9d66e7322195487eb687ab7)

File: app.py (hunk 7 above)
Comment (gdiazc) [reply app.py]: Ready! See [564fbce](https://github.com/fund3/sentiment-dashboard/pull/3/commits/564fbce1529837aff9d8f5d1000d3bb72073b28b)

**Review by dev1-fund3 — APPROVED** [review]

**Pull request merged**